### ISSUE TITLE
CryoEM

### ISSUE THREAD

**Issue by lockshaw** [flexflow/FlexFlow#787]
Moved from #642 

1. > @soumyac1999 
   > Things which need to be implemented:
   > 1. BatchNorm1D (#592)
   > 1. ~Regularization (#354)~ (#708)
   > 1. ~Modify dataloaders to support constants~ (local change in dataloader)
   > 1. ~Create constant or in general, better ways to handle constants (not passing them via the dataloader) (#588)~ (local dataloader changes make this almost okay)
   > 1. ~Operations involving constant scalars (#604)~ (local dataloader changes make this almost okay)
   >
   > Things that probably exist:
   > 1. ~Splitting and Slicing~ (use native python frontend)
   > 1. ~Getting predictions from models (check in pytorch alignment)~
   > 
   > Other items:
   > 1. Enabling search - currently disabled due to network topology not being a series–parallel graphs (#497)
   > 1. Printing custom loss values during training
   > 1. ~Reduce mean~ (#676)
   > 1. ~Adding noise by not modifying dataloader (ref: the description of #365)~ (skip)
   > 1. ~Dense for more than 2 dimensional inputs~ (handled by reshape)
   > 1. ~Better alternatives to `torch.gather` (the index inputs are huge). `torch.index_select` or `tf.gather`?~ (works for now)
